relay-event-schema/src/protocol/span.rs

```diff
@@ -86,7 +109,7 @@ pub struct Span {
     /// This shall move to a stable location once we have stabilized the
     /// interface.  This is intentionally not typed today.
     #[metastructure(skip_serialization = "empty")]
-    pub _metrics_summary: Annotated<Value>,
+    pub _metrics_summary: Annotated<Object<Array<MetricSummary>>>,
```

Comment:
This should be `Annotated<MetricsSummary>` similar to `Measurements`. `Measurements` also does some validation in the `from_value` which we could also use for `MetricsSummary`.

I think ideally it could look like:

```rs
pub struct MetricsSummary(Map<MetricResourceIdentifier, ...>)
```

With a proper `FromValue` and `IntoValue` implementation.